duckduckgo/macos-browser
# Put Sparkle behind a new SPARKLE flag

Task/Issue URL: https://app.asana.com/0/1199230911884351/1205484519205751/f
Tech Design URL:
CC: @diegoreymendez @ayoy 

**Description**:

This PR updates the Sparkle logic to use its own `SPARKLE` flag, rather than relying on `!APPSTORE`.

This is being done because we have cases where we sometimes want to publish internal non-App Store builds for testing that aren't checking for updates against the main update channel. NetP has done this, DBP is currently doing this, and NetP is about to do it for a second time.

**Steps to test this PR**:
1. Run the `DuckDuckGo Privacy Browser` target and check that the Check For Updates menu is available, and works
2. Run the `DuckDuckGo Privacy Browser App Store` target and check that the Check For Updates menu is NOT available
3. Run the `DuckDuckGo DBP` target and check that the Check For Updates menu is NOT available
4. Archive + notarize the `DuckDuckGo Privacy Browser` target and check that the Check For Updates menu is available



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
